packages/react-aria-components/docs/styling.mdx

```diff
@@ -0,0 +1,351 @@
+    border-radius: 0 0 4px 4px;
+  }
+}
+```
+
+The slots supported by each component are shown in the Anatomy section of their documentation.
+
+## CSS variables
+
+Some components provide CSS variables that you can use in your styling code. For example, the [Select](Select.html) component provides a `--trigger-width` variable on the popover that is set to the width of the trigger button. You can use this to make the width of the popover match the width of the button.
+
+```css render=false
+.react-aria-Popover {
+  width: var(--trigger-width);
+}
+```
+
+The CSS variables provided by each component are listed in the Styling section of their documentation.
+
+## Tailwind CSS
+
+[Tailwind CSS](https://tailwindcss.com) is a utility-first CSS framework for rapid styling that works great with React Aria Components. To access [states](#states), you can use [data attributes](https://tailwindcss.com/docs/hover-focus-and-other-states#data-attributes) as modifiers:
+
+```jsx
+<Item className="data-[selected]:bg-blue-400 data-[disabled]:bg-gray-100">
+  Item
+</Item>
+```
+
+Alternatively, you can use [render props](#render-props) to control which Tailwind classes are applied based on states. This can be useful if you need to apply multiple clases based on a single state:
+
+```jsx
+<Radio
+  className={({isFocusVisible, isSelected}) => `
+    flex rounded-lg p-4
+    ${isFocusVisible ? 'ring-2 ring-blue-600 ring-offset-1' : ''}
+    ${isSelected ? 'bg-blue-600 border-white/30 text-white' : ''}
+  `}>
+  {/* ... */}
+</Radio>
+```
+
+To access [CSS variables](#css-variables), use Tailwind's [arbitrary value](https://tailwindcss.com/docs/adding-custom-styles#using-arbitrary-values) syntax.
+
+```jsx
+<Popover className="w-[--trigger-width]">
+  {/* ... */}
+</Popover>
+```
+
+### Plugin
+
+A Tailwind CSS plugin is also available to make styling states of React Aria Components easier, with shorter names and autocomplete in your editor. To install:
+
+```
+yarn add tailwindcss-react-aria-components
+```
+
+Then add the plugin to your `tailwind.config.js` file:
+
+```jsx
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  plugins: [
+    require('tailwindcss-react-aria-components')
+  ]
+};
+```
+
+With the plugin installed, you can now access all states without the `data-` prefix. If you have the [Tailwind VSCode Extension](https://tailwindcss.com/docs/editor-setup#intelli-sense-for-vs-code) installed, you'll also get autocomplete for all states in your editor.
+
+```jsx
+<Item className="selected:bg-blue-400 disabled:bg-gray-100">
+  Item
+</Item>
+```
+
+### Boolean states
+
+Boolean states such as `data-pressed` can be styled with `pressed:` like this:
+
+```jsx
+<Button className="pressed:bg-blue">
+  {/* ... */}
+</Button>
+```
+
+### Non-boolean states
+
+Non-boolean states follow the `{name}-{value}` pattern, so you can style an element with `data-orientation="vertical"` using `orientation-vertical:`.
```

Comment:
```suggestion
Non-boolean states follow the `{name}-{value}` pattern, so you can style an element with `data-orientation="vertical"` using `orientation-vertical:`
```

Reply:
isn't the period correct? the colon is inside the backticks

Reply:
Ah you're right, just looked odd visually in the docs themselves. I'll revert
<img width="184" alt="image" src="https://github.com/adobe/react-spectrum/assets/9661127/12fa614b-0334-4022-8ba7-f49af4443a06">